File: inetpub/wwwroot/Services/Applications/AppChrysler/AppChrysler/MasterPage/AppMaster.Master.vb
Imports CKG.Base.Kernel
Imports CKG.Base.Kernel.Common.Common

Partial Public Class AppMaster
    Inherits System.Web.UI.MasterPage
    Private m_User As Security.User
    Protected Sub Page_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load

    End Sub

    Private Sub Page_PreRender(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.PreRender
        Dim strLogoPath As String = ""
        Dim strLogoPath2 As String = ""
        Dim strDocuPath As String = ""
        Dim strTitle As String
        Dim bc As HttpBrowserCapabilities
        bc = Request.Browser

        'Aktuelles Jahr ins Copyright setzen.
        lblCopyright.Text = lblCopyright.Text.Replace("year", DateTime.Now.Year.ToString)

        lnkLogout.NavigateUrl = "/Services/(S(" & Session.SessionID & "))/Start/Logout.aspx"
        m_User = CType(Session("objUser"), Security.User)
        If m_User Is Nothing Then
            If Not IsPostBack Then
                tdHandbuch.Visible = False
                lnkHauptmenue.Text = "Anmeldung"
                tdChangePasword.Visible = False
                'tdLogout.Visible = False
                lblUserName.Visible = False
                imgLogo.Visible = False
                PlaceHeader.Visible = True

            End If
        Else
            imgLogo.Visible = True
            PlaceHeader.Visible = False
            lnkHauptmenue.NavigateUrl = "/Services/(S(" & Session.SessionID & "))/Start/Selection.aspx"
            lnkChangePassword.NavigateUrl = "/Services/(S(" & Session.SessionID & "))/Start/ChangePassword.aspx"
            lnkLogout.NavigateUrl = "/Services/(S(" & Session.SessionID & "))/Start/Logout.aspx"
            lnkContact.NavigateUrl = "/Services/(S(" & Session.SessionID & "))/Info/ContactPage.aspx"
            lnkResponsible.NavigateUrl = "/Services/(S(" & Session.SessionID & "))/Info/ResponsiblePage.aspx"
            lnkImpressum.NavigateUrl = "/Services/(S(" & Session.SessionID & "))/Info/Impressum.aspx"

            PlaceHeader.Visible = False

            lnkChangePassword.Visible = True
            lblBenutzer.Text = "Benutzer: "
            lblUserName.Text = m_User.UserName
            lnkLogout.Visible = True
            lnkChangePassword.Visible = True

            Dim strCSSLink As String = ""
            With bc
                If .Type = "IE6" Then

                    strCSSLink = "<link href=""/Services/Styles/defaultIE6.css"" media=""screen, projection"" type=""text/css"" rel=""stylesheet"" />"
                    Select Case m_User.CustomerName
                        Case "Volksfürsorge"
                            strCSSLink &= "<link href=""Services/Customize/Wuerttenbergische/wuerttembIE6.css"" media=""screen, projection"" type=""text/css"" rel=""stylesheet"" />"
                        Case "AKF Bank Retail"
                            strCSSLink &= "<link href=""/Services/Customize/Akf_Retail/AKFRetailIE6.css"" media=""screen, projection"" type=""text/css"" rel=""stylesheet"" />"
                        Case "Arval"
                            strCSSLink &= "<link href=""/Services/Customize/Arval/ArvalIE6.css"" media=""screen, projection"" type=""text/css"" rel=""stylesheet"" />"
                        Case "Porsche"
                            strCSSLink &= "<link href=""/Services/Customize/porsche/porscheE6.css"" media=""screen, projection"" type=""text/css"" rel=""stylesheet"" />"
                        Case "Volvo Auto Bank GmbH"
                            strCSSLink &= "<link href=""/Services/Customize/Volvo/volvoIE6.css"" media=""screen, projection"" type=""text/css"" rel=""stylesheet"" />"
                        Case "Chrysler"
                            strCSSLink &= "<link href=""/Services/Customize/Chrysler/ChryslerIE6.css"" media=""screen, projection"" type=""text/css"" rel=""stylesheet"" />"

                        Case Else

                    End Select

                Else
                    Dim strtemp As String = Server.MapPath("~/Services/Styles/default.css")
                    strCSSLink = "<link href=""/Services/Styles/default.css"" media=""screen, projection"" type=""text/css"" rel=""stylesheet"" />"
                    Select Case m_User.CustomerName
                        Case "Volksfürsorge"
                            strCSSLink &= "<link href=""/Services/Customize/Wuerttenbergische/wuerttenb.css"" media=""screen, projection"" type=""text/css"" rel=""stylesheet"" />"
                        Case "AKF Bank Retail"
                            strCSSLink &= "<link href=""/Services/Customize/Akf_Retail/AKFRetail.css"" media=""screen, projection"" type=""text/css"" rel=""stylesheet"" />"
                        Case "Arval"
                            strCSSLink &= "<link href=""/Services/Customize/Arval/Arval.css"" media=""screen, projection"" type=""text/css"" rel=""stylesheet"" />"
                        Case "Porsche"
                            strCSSLink &= "<link href=""/Services/Customize/porsche/porsche.css"" media=""screen, projection"" type=""text/css"" rel=""stylesheet"" />"
                        Case Else
                            strCSSLink &= "<link href=""" & m_User.Customer.CustomerStyle.CssPath & """ media=""screen, projection"" type=""text/css"" rel=""stylesheet"" />"
                    End Select
                End If
            End With
            Me.Head1.Controls.Add(New LiteralControl(strCSSLink))

            If Me.Page.User.Identity.IsAuthenticated Then
                Select Case Me.Page.Title
                    Case "Startseite"
                        tdHauptmenue.Attributes.Add("class", "active")
                        tdChangePasword.Attributes.Add("class", "")
                        'tdLogout.Attributes.Add("class", "")
                        tdContact.Attributes.Add("class", "")
                        tdImpressum.Attributes.Add("class", "")
                        tdResponsible.Attributes.Add("class", "")

                    Case "Passwort ändern"
                        tdHauptmenue.Attributes.Add("class", "")
                        tdChangePasword.Attributes.Add("class", "active")
                        'tdLogout.Attributes.Add("class", "")
                        tdContact.Attributes.Add("class", "")
                        tdImpressum.Attributes.Add("class", "")
                        tdResponsible.Attributes.Add("class", "")
                    Case "Kontaktseite"
                        tdHauptmenue.Attributes.Add("class", "")
                        tdChangePasword.Attributes.Add("class", "")
                        ' tdLogout.Attributes.Add("class", "")
                        tdContact.Attributes.Add("class", "active")
                        tdImpressum.Attributes.Add("class", "")
                        tdResponsible.Attributes.Add("class", "")
                    Case "Impressum"
                        tdHauptmenue.Attributes.Add("class", "")
                        tdChangePasword.Attributes.Add("class", "")
                        'tdLogout.Attributes.Add("class", "")
                        tdContact.Attributes.Add("class", "")
                        tdImpressum.Attributes.Add("class", "active")
                        tdResponsible.Attributes.Add("class", "")
                    Case "Ansprechpartner"
                        tdHauptmenue.Attributes.Add("class", "")
                        tdChangePasword.Attributes.Add("class", "")
                        'tdLogout.Attributes.Add("class", "")
                        tdContact.Attributes.Add("class", "")
                        tdImpressum.Attributes.Add("class", "")
                        tdResponsible.Attributes.Add("class", "active")
                    Case Else
                        tdHauptmenue.Attributes.Add("class", "")
                        tdChangePasword.Attributes.Add("class", "")
                        'tdLogout.Attributes.Add("class", "")
                        tdContact.Attributes.Add("class", "")
                        tdImpressum.Attributes.Add("class", "")
                        tdResponsible.Attributes.Add("class", "")
                End Select

                If Me.Page.User.Identity.IsAuthenticated Then
                    If Me.Page.Title = "Startseite" Then
                        strTitle = m_User.Customer.CustomerName & " - " & "Startseite"
                        Me.Page.Title = strTitle

                    ElseIf Session("AppID") Is Nothing Then
                        Me.Page.Title = m_User.Customer.CustomerName

                    Else
                        strTitle = m_User.Applications.Select("AppID = '" & Session("AppID").ToString & "'")(0)("AppFriendlyName").ToString
                        Me.Page.Title = m_User.Customer.CustomerName & " - " & strTitle

                    End If

                End If
            End If

            If m_User.GroupID > 0 Then
                strLogoPath = m_User.Organization.LogoPath
                strDocuPath = m_User.Groups.ItemByID(m_User.GroupID).DocuPath

                Dim cn As SqlClient.SqlConnection
                cn = New SqlClient.SqlConnection(m_User.App.Connectionstring)
                Dim _EmployeeAssigned As New Admin.EmployeeList(m_User.GroupID, m_User.Customer.AccountingArea, cn)
                _EmployeeAssigned.GetAssigned()
                If _EmployeeAssigned.DefaultView.Count > 0 Then
                    _EmployeeAssigned.DefaultView.Sort = "EmployeeName"
                    tdResponsible.Visible = True
                Else
                    tdResponsible.Visible = False
                End If
            End If
            If strDocuPath = String.Empty Then
                strDocuPath = m_User.Customer.DocuPath
            End If
            If Not strDocuPath = String.Empty Then
                tdHandbuch.Visible = True
                lnkHandbuch.NavigateUrl = strDocuPath
            End If

            If strLogoPath = String.Empty Then
                strLogoPath = m_User.Customer.CustomerStyle.LogoPath
            End If

        End If
    End Sub
End Class

File: inetpub/wwwroot/Services/Applications/AppMBB/master/AppMaster.Master.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Text.RegularExpressions;
using CKG.Base.Kernel;
using CKG.Base.Kernel.Common;
using CKG.Services;

namespace AppMBB.Master
{
    public partial class AppMaster : System.Web.UI.MasterPage
    {
        private CKG.Base.Kernel.Security.User m_User;


        protected void Page_Load(object sender, EventArgs e)
        {

        }
        protected void Page_PreRender(object sender, EventArgs e)
        {

            string strLogoPath = "";
            string strDocuPath = "";
            string strTitle = null;
            HttpBrowserCapabilities bc = default(HttpBrowserCapabilities);
            bc = Request.Browser;

            //Aktuelles Jahr ins Copyright setzen.
            lblCopyright.Text = lblCopyright.Text.Replace("year", DateTime.Now.Year.ToString());

            lnkLogout.NavigateUrl = "/Services/(S(" + Session.SessionID + "))/Start/Logout.aspx";
            m_User = (CKG.Base.Kernel.Security.User)Session["objUser"];
            if (m_User == null)
            {
                if (!IsPostBack)
                {
                    tdHandbuch.Visible = false;
                    lnkHauptmenue.Text = "Anmeldung";
                    tdChangePasword.Visible = false;
                    //tdLogout.Visible = False 
                    lblUserName.Visible = false;
                    imgLogo.Visible = false;

                    PlaceHeader.Visible = true;
                }
            }
            else
            {
                imgLogo.Visible = true;
                PlaceHeader.Visible = false;
                lnkHauptmenue.NavigateUrl = "/Services/(S(" + Session.SessionID + "))/Start/Selection.aspx";
                lnkChangePassword.NavigateUrl = "/Services/(S(" + Session.SessionID + "))/Start/ChangePassword.aspx";
                lnkLogout.NavigateUrl = "/Services/(S(" + Session.SessionID + "))/Start/Logout.aspx";
                lnkContact.NavigateUrl = "/Services/(S(" + Session.SessionID + "))/Info/ContactPage.aspx";
                lnkResponsible.NavigateUrl = "/Services/(S(" + Session.SessionID + "))/Info/ResponsiblePage.aspx";
                lnkImpressum.NavigateUrl = "/Services/(S(" + Session.SessionID + "))/Info/Impressum.aspx";

                PlaceHeader.Visible = false;

                lnkChangePassword.Visible = true;
                lblBenutzer.Text = "Benutzer: ";
                lblUserName.Text = m_User.UserName;
                lnkLogout.Visible = true;
                lnkChangePassword.Visible = true;

                string strCSSLink = "";
                {
                    if (bc.Type == "IE6")
                    {

                        strCSSLink = "<link href=\"/Services/Styles/defaultIE6.css\" media=\"screen, projection\" type=\"text/css\" rel=\"stylesheet\" />";

                        string[] strCSS = null;
                        string strCSSPath = m_User.Customer.CustomerStyle.CssPath;
                        if (strCSSPath.Contains(".css"))
                        {


                            strCSS = Regex.Split(strCSSPath, ".css");

                            if (strCSS.Length == 2)
                            {
                                strCSSPath = strCSS[0] + "IE6.css";
                                strCSSLink += "<link href=\"" + strCSSPath + "\" media=\"screen, projection\" type=\"text/css\" rel=\"stylesheet\" />";
                            }
                        }
                    }
                    else
                    {
                        string strtemp = Server.MapPath("~/Services/Styles/default.css");
                        strCSSLink = "<link href=\"/Services/Styles/default.css\" media=\"screen, projection\" type=\"text/css\" rel=\"stylesheet\" />";
                        switch (m_User.CustomerName)
                        {
                            case "Volksfürsorge":
                                strCSSLink += "<link href=\"/Services/Customize/Wuerttenbergische/wuerttenb.css\" media=\"screen, projection\" type=\"text/css\" rel=\"stylesheet\" />";
                                break;
                            case "AKF Bank Retail":
                                strCSSLink += "<link href=\"/Services/Customize/Akf_Retail/AKFRetail.css\" media=\"screen, projection\" type=\"text/css\" rel=\"stylesheet\" />";
                                break;
                            case "Arval":
                                strCSSLink += "<link href=\"/Services/Customize/Arval/Arval.css\" media=\"screen, projection\" type=\"text/css\" rel=\"stylesheet\" />";
                                break;
                            case "Porsche":
                                strCSSLink += "<link href=\"/Services/Customize/porsche/porsche.css\" media=\"screen, projection\" type=\"text/css\" rel=\"stylesheet\" />";
                                break;
                            default:
                                strCSSLink += "<link href=\"" + m_User.Customer.CustomerStyle.CssPath + "\" media=\"screen, projection\" type=\"text/css\" rel=\"stylesheet\" />";
                                break;
                        }
                    }
                }
                this.Head1.Controls.Add(new LiteralControl(strCSSLink));

                if ((HttpContext.Current.Request.UserAgent != null) && (HttpContext.Current.Request.UserAgent.ToLower().Contains("msie 10")))
                {
                    this.Head1.Controls.AddAt(0, new LiteralControl("<META content=\"IE=9,chrome=1\" http-equiv=\"X-UA-Compatible\">"));
                }

                if (this.Page.User.Identity.IsAuthenticated)
                {
                    switch (this.Page.Title)
                    {
                        case "Startseite":
                            tdHauptmenue.Attributes.Add("class", "active");
                            tdChangePasword.Attributes.Add("class", "");
                            //tdLogout.Attributes.Add("class", "") 
                            tdContact.Attributes.Add("class", "");
                            tdImpressum.Attributes.Add("class", "");
                            tdResponsible.Attributes.Add("class", "");

                            break;
                        case "Passwort ändern":
                            tdHauptmenue.Attributes.Add("class", "");
                            tdChangePasword.Attributes.Add("class", "active");
                            //tdLogout.Attributes.Add("class", "") 
                            tdContact.Attributes.Add("class", "");
                            tdImpressum.Attributes.Add("class", "");
                            tdResponsible.Attributes.Add("class", "");
                            break;
                        case "Kontaktseite":
                            tdHauptmenue.Attributes.Add("class", "");
                            tdChangePasword.Attributes.Add("class", "");
                            // tdLogout.Attributes.Add("class", "") 
                            tdContact.Attributes.Add("class", "active");
                            tdImpressum.Attributes.Add("class", "");
                            tdResponsible.Attributes.Add("class", "");
                            break;
                        case "Impressum":
                            tdHauptmenue.Attributes.Add("class", "");
                            tdChangePasword.Attributes.Add("class", "");
                            //tdLogout.Attributes.Add("class", "") 
                            tdContact.Attributes.Add("class", "");
                            tdImpressum.Attributes.Add("class", "active");
                            tdResponsible.Attributes.Add("class", "");
                            break;
                        case "Ansprechpartner":
                            tdHauptmenue.Attributes.Add("class", "");
                            tdChangePasword.Attributes.Add("class", "");
                            //tdLogout.Attributes.Add("class", "") 
                            tdContact.Attributes.Add("class", "");
                            tdImpressum.Attributes.Add("class", "");
                            tdResponsible.Attributes.Add("class", "active");
                            break;
                        default:
                            tdHauptmenue.Attributes.Add("class", "");
                            tdChangePasword.Attributes.Add("class", "");
                            //tdLogout.Attributes.Add("class", "") 
                            tdContact.Attributes.Add("class", "");
                            tdImpressum.Attributes.Add("class", "");
                            tdResponsible.Attributes.Add("class", "");
                            break;
                    }

                    if (this.Page.User.Identity.IsAuthenticated)
                    {
                        if (this.Page.Title == "Startseite")
                        {
                            strTitle = m_User.Customer.CustomerName + " - " + "Startseite";

                            this.Page.Title = strTitle;
                        }
                        else if (Session["AppID"] == null)
                        {

                            this.Page.Title = m_User.Customer.CustomerName;
                        }
                        else
                        {
                            strTitle = (string)m_User.Applications.Select("AppID = '" + Session["AppID"] + "'")[0]["AppFriendlyName"];


                            this.Page.Title = m_User.Customer.CustomerName + " - " + strTitle;

                        }
                    }
                }

                if (m_User.GroupID > 0)
                {
                    strLogoPath = m_User.Organization.LogoPath;
                    strDocuPath = m_User.Groups.get_ItemByID(m_User.GroupID).DocuPath;


                    System.Data.SqlClient.SqlConnection cn = default(System.Data.SqlClient.SqlConnection);
                    cn = new System.Data.SqlClient.SqlConnection(m_User.App.Connectionstring);

                    CKG.Base.Kernel.Admin.EmployeeList _EmployeeAssigned = new CKG.Base.Kernel.Admin.EmployeeList(m_User.GroupID, m_User.Customer.AccountingArea, cn);
                    _EmployeeAssigned.GetAssigned();
                    if (_EmployeeAssigned.DefaultView.Count > 0)
                    {
                        _EmployeeAssigned.DefaultView.Sort = "EmployeeName";
                        tdResponsible.Visible = true;
                    }
                    else
                    {
                        tdResponsible.Visible = false;
                    }
                }
                if (strDocuPath == string.Empty)
                {
                    strDocuPath = m_User.Customer.DocuPath;
                }
                if (!(strDocuPath == string.Empty))
                {
                    tdHandbuch.Visible = true;
                    lnkHandbuch.NavigateUrl = strDocuPath;
                }

                if (strLogoPath == string.Empty)
                {
                    strLogoPath = m_User.Customer.CustomerStyle.LogoPath;

                }
            }

            Services ServicesHeader = new Services();
            ServicesHeader.FillHeaderMitKundenlogo(ref imgLogo, m_User, ref JavaScriptBlock, ref imgLogoDIV);
        }

    }
}
